### PR TITLE
command crd

### PR DESCRIPTION
**Pull request by amirmalka**

### CODE REVIEW

File: pkg/command/types/v1alpha1/types.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+type OperatorCommandSpec struct {
+	GUID           string `json:"guid"`                     // GUID is a unique identifier for the command
+	CommandType    string `json:"commandType"`              // CommandType is the type of the command
+	CommandVersion string `json:"commandVersion,omitempty"` // CommandVersion is the version of the command
+	// labels should include - command type, node name, app name (node-agent, operator,kubescape, ..)
+	Designators []identifiers.PortalDesignator `json:"designators,omitempty"` // Designators are the designators for the command (on which the command will be executed)
+	// TODO: Body - (suggestion to use gob) https://pkg.go.dev/encoding/gob
+	// TODO: Add TTL
+	Args         map[string]interface{} `json:"args,omitempty"`         // Args are the arguments for the command
```
Comment (amirmalka) [inline pkg/command/types/v1alpha1/types.go]: Should be something like  `Body` of type `[]byte`
https://pkg.go.dev/encoding/gob

File: pkg/command/crds/crd.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (amirmalka) [inline pkg/command/crds/crd.yaml]:
```
apiVersion: kubescape.io/v1alpha1
kind: OperatorCommand
metadata:
  name: example-operator-command
spec:
  guid: "123e4567-e89b-12d3-a456-426614174000"
  commandType: "Scan"
  designators: "cluster"
  args:
    namespace: "default"
    scanProfile: "strict"
status:
  started: true
  startedAt: "2024-09-10T12:34:56Z"
  completed: false
  error:
    reason: ""
    message: ""
    errorCode: 0

```

Comment (jnathangreeg) [reply pkg/command/crds/crd.yaml:1]: Need to add node name(Amit)

File: pkg/command/crds/crd-status-subresource.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                    reason:
+                      type: string
+                started:
+                  type: boolean
+                startedAt:
+                  type: string
+                  format: date-time
+      subresources:
+        status: {}
+  conversion:
```
Comment (jnathangreeg) [inline pkg/command/crds/crd-status-subresource.yaml]: remove

File: pkg/command/types/v1alpha1/types.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+	Spec   OperatorCommandSpec   `json:"spec,omitempty"`
+	Status OperatorCommandStatus `json:"status,omitempty"`
+}
+
+type OperatorCommandSpec struct {
+	GUID           string `json:"guid"`                     // GUID is a unique identifier for the command
+	CommandType    string `json:"commandType"`              // CommandType is the type of the command
+	CommandVersion string `json:"commandVersion,omitempty"` // CommandVersion is the version of the command
+	Labels         map[string]string              `json:"labels,omitempty"`       // Labels are the labels for the command
```
Comment (jnathangreeg) [inline pkg/command/types/v1alpha1/types.go]: remove its part of the meta

File: pkg/command/crds/crd.yaml (hunk 2 above)
Comment (jnathangreeg) [reply pkg/command/crds/crd.yaml]: apiVersion: kubescape.io/v1alpha1
kind: OperatorCommand
metadata:
  name: new-operatorcommand  
spec:
  guid: "123e4567-e89b-12d3-a456-426614174000"
  commandType: "ExecutePolicy"
  commandVersion: "1.0"
  designators:
    - key: "cluster"
      value: "arn-aws-eks-eu-west-1-015253967648-cluster-ca-terraform-eks-dev-stage"
  body: "eyJrZXkiOiJ2YWx1ZSJ9"
  ttl: "1h"
  args:
    policyName: "SecurityPolicy"
    policyLevel: "high"
  commandIndex: 1
  commandCount: 5
status:
  started: true
  startedAt: "2024-04-09T11:03:29Z"
  completed: false
  executer: "admin"
  error:
    reason: ""
    message: ""
    errorCode: 0

**Review by jnathangreeg — APPROVED** [review]

**Pull request merged**